mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py

```diff
@@ -144,6 +154,7 @@ def finish(self):
                     self.selected_resource.closeConnection()
                 if self.selected_resource.is_allocated:
                     self.selected_resource.release()
+                    self.selected_resource = None
```

Comment:
@jupe The whole "this function is called twice" I think is just for completeness, not design. It's called in the destructor in the off chance you haven't cleaned up the connection cleanly.

Sounds like this statement belongs where it is 👍